server/text_generation/utils/watermark.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+# Copyright 2023 Authors of "A Watermark for Large Language Models"
+# available at https://arxiv.org/abs/2301.10226
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from transformers import LogitsProcessor
+
+
+class WatermarkLogitsProcessor(LogitsProcessor):
+    def __init__(
+        self,
+        vocab_size: int,
+        gamma: float = 0.5,
```

Comment:
All these parameters need to be fixed for the detector to work right?

Reply:
Yes, or rather _they need to match at detection time_. The core assumption is realized by the class structure. 

The base class logic, parametrized by kwargs, is the contract between the processor (watermark-_er_) and the detector. If they don't match, then the detector will (in expectation) return False/unwatermarked, since the pseudorandom procedure won't "line up" and thus the outputs will read as unwatermarked.